### PR TITLE
Make getKubeconfig method available to multi-lang

### LINKED ISSUE #613: Expose methods to multi-lang
At a minimum, we should expose [`Cluster.getKubeConfig`](https://github.com/pulumi/pulumi-eks/blob/86861e6cb6091e97d08828a444bb70087a8701f3/nodejs/eks/cluster.ts#L1482), and time-permitting [`Cluster.createNodeGroup`](https://github.com/pulumi/pulumi-eks/blob/86861e6cb6091e97d08828a444bb70087a8701f3/nodejs/eks/cluster.ts#L1453). (The latter is nice-to-have as it's just a convenience over the `NodeGroup` component, but still would be a good demonstration of the methods feature.)

The work involves:

 - Adopting the latest codegen (ideally with support for https://github.com/pulumi/pulumi/issues/7435)
 - Adding the methods to the schema
 - Hooking up the methods in the provider

### PR DESCRIPTION
**Pull request by justinvp**
Fixes #613

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,7 +38,7 @@ build_python:: schema
 		cp ../README.md . && \
 		python3 setup.py clean --all 2>/dev/null && \
 		rm -rf ./bin/ ../python.bin/ && cp -R . ../python.bin && mv ../python.bin ./bin && \
-		sed -i.bak -e "s/\$${VERSION}/$(PYPI_VERSION)/g" -e "s/\$${PLUGIN_VERSION}/$(VERSION)/g" ./bin/setup.py && \
+		sed -i.bak -e 's/^VERSION = .*/VERSION = "$(PYPI_VERSION)"/g' -e 's/^PLUGIN_VERSION = .*/PLUGIN_VERSION = "$(VERSION)"/g' ./bin/setup.py && \
```
Comment (lblackstone) [inline Makefile:41]: Curious why this change was needed?

File: examples/examples_py_test.go [hunk 2]
```diff
@@ -39,6 +39,22 @@ func TestAccAwsProfilePy(t *testing.T) {
 	integration.ProgramTest(t, &test)
 }
 
+func TestAccAwsProfileRolePy(t *testing.T) {
+	test := getPythonBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir: path.Join(getCwd(t), "aws-profile-role-py"),
+			// TODO: Temporarily skip the extra runtime validation due to test failure.
```
Comment (lblackstone) [inline examples/examples_py_test.go:46]: Is there a GH issue tracking this?

**Review by lblackstone — APPROVED** [review]

File: examples/examples_py_test.go (hunk 2 above)
Comment (emiliza) [reply examples/examples_py_test.go:46]: lol, oh, looks like this has been a known issue seeing it commented out in a bunch of other tests... :(

File: Makefile (hunk 1 above)
Comment (emiliza) [reply Makefile:41]: Justin made a similar change a few weeks ago in `pulumi-kubernetes` - check the rationale here: https://github.com/pulumi/pulumi-kubernetes/pull/1664
